### PR TITLE
PUBDEV-7425: Make GAM parameter names more descriptive

### PR DESCRIPTION
**Pull request by wendycwong**
This PR completes the task in JIRA: https://0xdata.atlassian.net/browse/PUBDEV-7425

I was blind and did not see the code review comments from Erins to make the GAM parameters more meaningful.

I changed parameters:
- k to num_knots.
- save_gam_cols to keep_gam_cols
- knots_keys to knot_ids
- gam_x to gam_columns

I added fractionalbinomial to gam supported families.

### CODE REVIEW

**Comment by ledell** [comment]
@angela0xdata Is there a corresponding User Guide PR for the variable name changes?

**Comment by angela0xdata** [comment]
@ledell yes: https://github.com/h2oai/h2o-3/pull/4484 ... and i think it's ready for review

File: h2o-py/h2o/estimators/gam.py [hunk 1]
```diff
@@ -860,48 +861,48 @@ def custom_metric_func(self, custom_metric_func):
...
+    def num_knots(self, num_knots):
+        assert_is_type(num_knots, None, [int])
+        self._parms["num_knots"] = num_knots
 
 
     @property
-    def knots_keys(self):
+    def knot_ids(self):
         """
         String arrays storing frame keys of knots.  One for each gam column specified in gam_x
```
Comment (angela0xdata) [inline h2o-py/h2o/estimators/gam.py:881]: >>String arrays storing frame keys of knots.  One for each gam column specified in gam_x
Change to
String arrays storing frame keys of knots.  One for each GAM column specified in gam_columns

File: h2o-r/h2o-package/R/gam.R [hunk 2]
```diff
@@ -103,11 +102,12 @@
 #'        Defaults to 0.
 #' @param max_runtime_secs Maximum allowed runtime in seconds for model training. Use 0 to disable. Defaults to 0.
 #' @param custom_metric_func Reference to custom evaluation function, format: `language:keyName=funcName`
-#' @param k Number of knots for gam predictors
-#' @param knots_keys String arrays storing frame keys of knots.  One for each gam column specified in gam_x
+#' @param num_knots Number of knots for gam predictors
+#' @param knot_ids String arrays storing frame keys of knots.  One for each gam column specified in gam_x
```
Comment (angela0xdata) [inline h2o-r/h2o-package/R/gam.R:106]: same here:
One for each GAM column specified in gam_columns

File: h2o-algos/src/main/java/hex/schemas/GAMV3.java [hunk 3]
```diff
@@ -232,10 +232,10 @@
     //public BSType bs;
 
     @API(help="Save keys of model matrix", level = Level.secondary, direction = Direction.INPUT)
-    public boolean save_gam_cols; // if true will save keys storing GAM columns
+    public boolean keep_gam_cols; // if true will save keys storing GAM columns
     
     @API(help="String arrays storing frame keys of knots.  One for each gam column specified in gam_x",
```
Comment (angela0xdata) [inline h2o-algos/src/main/java/hex/schemas/GAMV3.java:237]: Last one to change here.
"String arrays storing frame keys of knots.  One for each GAM column specified in gam_columns",

Comment (wendycwong) [reply h2o-algos/src/main/java/hex/schemas/GAMV3.java:237]: Good catch Angela.  Fixed

File: h2o-r/h2o-package/R/gam.R (hunk 2 above)
Comment (wendycwong) [reply h2o-r/h2o-package/R/gam.R]: Fixed that too!

**Comment by Pscheidl** [comment]
Waiting for tests, then I'll merge to master and run the nightly tests in order to prepare for release.

**Comment by Pscheidl** [comment]
Plays well with master, ran the tests at least locally. It is expected not to create any problems after merge.

**Pull request merged**